src/databricks/labs/ucx/assessment/aws.py

```diff
@@ -0,0 +1,259 @@
+                logger.error(error)
+                return []
+            policy_version = json.loads(output)["Policy"]["DefaultVersionId"]
+            get_policy = (
+                f"aws iam get-policy-version --profile {self._profile} --policy-arn {attached_policy_arn} "
+                f"--version-id {policy_version} --no-paginate"
+            )
+        else:
+            logger.error("Failed to retrieve role. No role name or attached role ARN specified.")
+            return []
+        code, output, error = run_command(get_policy)
+        if code != 0:
+            logger.error(error)
+            return []
+        policy = json.loads(output)
+        if policy_name:
+            actions = policy["PolicyDocument"].get("Statement", [])
+        else:
+            actions = policy["PolicyVersion"]["Document"].get("Statement", [])
+        for action in actions:
+            if action.get("Effect", "Deny") != "Allow":
+                continue
+            actions = action["Action"]
+            s3_action = []
+            if isinstance(actions, list):
+                for single_action in actions:
+                    if single_action in self.S3_ACTIONS:
+                        s3_action.append(single_action)
+                        continue
+            elif actions in self.S3_ACTIONS:
+                s3_action = [actions]
+
+            if not s3_action:
+                continue
+            for resource in action.get("Resource", []):
+                match = re.match(self.S3_REGEX, resource)
+                if match:
+                    yield AWSPolicyAction("s3", set(s3_action), match.group(1))
+
+
+class AWSInstanceProfileCrawler(CrawlerBase[AWSInstanceProfile]):
+    def __init__(self, ws: WorkspaceClient, sbe: SqlBackend, schema):
+        super().__init__(sbe, "hive_metastore", schema, "aws_instance_profile", AWSInstanceProfile)
+        self._ws = ws
+        self._schema = schema
+
+    def _crawl(self) -> Iterable[AWSInstanceProfile]:
+        instance_profiles = self._ws.instance_profiles.list()
+        for instance_profile in instance_profiles:
+            yield AWSInstanceProfile(instance_profile.instance_profile_arn, instance_profile.iam_role_arn)
+
+    def snapshot(self) -> Iterable[AWSInstanceProfile]:
+        return self._snapshot(self._try_fetch, self._crawl)
+
+    def _try_fetch(self) -> Iterable[AWSInstanceProfile]:
+        for row in self._fetch(f"SELECT * FROM {self._schema}.{self._table}"):
+            yield AWSInstanceProfile(*row)
+
+
+class AWSResourcePermissions:
+    def __init__(
+        self,
+        ws: WorkspaceClient,
+        awsrm: AWSResources,
+        instance_profile_crawler: AWSInstanceProfileCrawler,
+        folder: str | None = None,
+    ):
+        if not folder:
+            folder = f"/Users/{ws.current_user.me().user_name}/.ucx"
+        self._folder = folder
+        self._awsrm = awsrm
+        self._instance_profile_crawler = instance_profile_crawler
```

Comment:
again, we don't need crawler! it's faster to call the API directly (few milliseconds) every time we need it. don't overcomplicate this

Reply:
Fixed